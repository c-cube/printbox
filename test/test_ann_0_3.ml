
let b =
  let open PrintBox in
  let open Printbox_unicode in
  PrintBox_unicode.setup();
  frame @@ grid_l [
    [text "subject"; text_with_style Style.bold "announce: printbox 0.3"];
    [text "explanation";
    frame @@ text {|PrintBox is a library for rendering nested tables,
    trees, and similar structures in monospace text or HTML.|}];
    [text "github";
    text_with_style Style.(bg_color Blue) "https://github.com/c-cube/printbox/releases/tag/0.3"];
    [text "contributors";
     vlist_map (text_with_style Style.(fg_color Green)) ["Simon"; "Guillaume"; "Matt"]];
    [text "dependencies";
    tree empty
      [tree (text "mandatory")
         [text "dune"; text "bytes"];
       tree (text "optional")
         [text "uutf"; text "uucp"; text "tyxml"]]];
    [text "expected reaction"; text "ðŸŽ‰"];
  ]

let () = print_endline @@ PrintBox_text.to_string b
