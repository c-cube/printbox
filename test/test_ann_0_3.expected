┌─────────────────┬──────────────────────────────────────────────────────────────┐
│subject          │[1mannounce: printbox 0.3[0m                                        │
├─────────────────┼──────────────────────────────────────────────────────────────┤
│explanation      │┌────────────────────────────────────────────────────────────┐│
│                 ││PrintBox is a library for rendering nested tables,          ││
│                 ││    trees, and similar structures in monospace text or HTML.││
│                 │└────────────────────────────────────────────────────────────┘│
├─────────────────┼──────────────────────────────────────────────────────────────┤
│github           │[44mhttps://github.com/c-cube/printbox/releases/tag/0.3[0m           │
├─────────────────┼──────────────────────────────────────────────────────────────┤
│contributors     │[32mSimon[0m                                                         │
│                 ├──────────────────────────────────────────────────────────────┤
│                 │[32mGuillaume[0m                                                     │
│                 ├──────────────────────────────────────────────────────────────┤
│                 │[32mMatt[0m                                                          │
├─────────────────┼──────────────────────────────────────────────────────────────┤
│dependencies     │┬─mandatory                                                   │
│                 ││ ├─dune                                                      │
│                 ││ ├─bytes                                                     │
│                 ││ ├─uutf                                                      │
│                 ││ └─uucp                                                      │
│                 │└─optional                                                    │
│                 │  └─tyxml                                                     │
├─────────────────┼──────────────────────────────────────────────────────────────┤
│expected reaction│🎉                                                            │
└─────────────────┴──────────────────────────────────────────────────────────────┘
